app/services/vacancy_facets.rb

```diff
@@ -0,0 +1,66 @@
+class VacancyFacets
+  FIELDS = %i[job_roles subjects cities counties].freeze
+
+  def initialize(store: Redis.new)
+    @store = store
+  end
+
+  def get(field)
+    return {} if !FIELDS.include?(field) || !store.exists?(field)
+
+    JSON.parse(store.get(field))
+  end
+
+  def refresh
+    set_job_role_facets
+    set_subjects_facets
+    set_cities_facets
+    set_counties_facets
+  end
+
+private
+
+  attr_reader :store
+
+  def set_job_role_facets
+    store.set(:job_roles, job_role_facet(Vacancy::JOB_ROLE_OPTIONS).to_json)
+  end
+
+  def set_subjects_facets
+    store.set(:subjects, subject_facet(SUBJECT_OPTIONS).to_json)
+  end
+
+  def set_cities_facets
+    store.set(:cities, location_category_facet(CITIES).to_json)
+  end
+
+  def set_counties_facets
+    store.set(:counties, location_category_facet(COUNTIES).to_json)
+  end
+
+  def job_role_facet(job_roles)
+    job_roles.filter_map { |job_role|
+      count = algolia_facet_count(job_roles: job_role)
+      [job_role.first, count] if count.positive?
+    }.to_h
+  end
+
+  def subject_facet(subjects)
+    subjects.filter_map { |subject|
+      count = algolia_facet_count(keyword: subject.first)
+      [subject.first, count] if count.positive?
+    }.to_h
+  end
+
+  def location_category_facet(locations)
+    locations.filter_map { |location|
+      count = algolia_facet_count(location: location, location_category: location, radius: 10)
```

Comment:
You should be able to provide just `location: location` or `location_category: location` only